websites/E/E-PAL/presence.ts

```diff
@@ -0,0 +1,230 @@
+			break;
+		}
+		case pathname.includes("/topic/"): {
+			presenceData.details = `Viewing ${epalName}'s post`;
+			presenceData.smallImageKey = epalAvatar ?? Assets.Logo;
+			presenceData.buttons = [{ label: "View Post", url: href }];
+			break;
+		}
+		case pathname.includes("/foryou"): {
+			presenceData.details = "Reading about";
+			presenceData.state = document.querySelector(
+				'div[class*="text-txt-primary-normal"]'
+			).textContent;
+			presenceData.smallImageKey = Assets.Reading;
+			break;
+		}
+		case pathname.includes("/trade/"): {
+			presenceData.details = "Viewing an order";
+			presenceData.state = `From - ${
+				document.querySelector('[class*="txt-title-s"]')?.textContent
+			}`;
+			presenceData.largeImageKey =
+				document
+					.querySelector('[class="flex items-center flex-1 min-w-0"]')
+					?.querySelector("img")
+					?.getAttribute("src")
+					?.split("?")[0] ?? Assets.Logo;
+			presenceData.smallImageKey =
+				document
+					.querySelector('[class="w-24"]')
+					.querySelector("img")
+					?.getAttribute("src") ?? "";
+			presenceData.buttons = [{ label: "View E-Pal", url: href }];
+			break;
+		}
+		case pathname.includes("/trade"): {
+			presenceData.details = `Viewing ${
+				document
+					.querySelector(
+						'[class="epal-menu-item bg-surface-element-normal rounded-m"]'
+					)
+					?.textContent?.toLowerCase() ?? "all"
+			} orders`;
+
+			break;
+		}
+		case pathname.includes("/epals/"): {
+			if (pathSplit[2]) {
+				const filters = document.querySelectorAll(
+						'[class*="epal-chip-active"]'
+					),
+					afilters = [];
+				for (const [, element] of filters.entries()) {
+					if (!element.textContent.includes("filters"))
+						afilters.push(element.textContent);
+				}
+				presenceData.details = `${
+					document.querySelector('[class="txt-headline-l"]')?.textContent ??
+					pathSplit[2]
+				} E-Pals`;
+				presenceData.state =
+					afilters.length === 1
+						? afilters[0]
+						: afilters.length > 1
+						? afilters.toString().replace(/,/gm, ", ")
+						: "";
+			}
+			break;
+		}
+		case pathname.includes("/faq"): {
+			presenceData.details = "Reading FAQ";
+			presenceData.state = document.querySelector(
+				'[class*="txt-title-l"]'
+			)?.textContent;
+			presenceData.smallImageKey = Assets.Reading;
+			presenceData.buttons = [{ label: "Read FAQ", url: href }];
+			break;
+		}
+		case pathname.includes("/coupon/"): {
+			presenceData.details = `Viewing their ${
+				document
+					.querySelector(
+						'[class="epal-menu-item bg-surface-element-normal rounded-m"]'
+					)
+					?.textContent?.toLowerCase() ?? "all"
+			} coupons`;
+			break;
+		}
+
+		default: {
+			if (Object.keys(staticPages).includes(pathname))
+				presenceData = { ...presenceData, ...staticPages[pathname] };
+			else presenceData.details = "Unknown page.";
+		}
+	}
+
+	if (presenceData.smallImageKey === "") delete presenceData.smallImageKey;
```

Comment:
This is due to an [alpha bug](https://discord.com/channels/493130730549805057/1160577657478590514) [https://discord.com/channels/493130730549805057/1160577657478590514](https://discord.com/channels/493130730549805057/1160577657478590514)